### ISSUE TITLE
Commerce & Partnerships Implementation (Phase 3) 

### ISSUE THREAD

**Issue by willykaram** [NYC-Camp/website#224]
As the third phase of the Commerce Implementation we want to add a bit more functionality, some of which overlaps with Partners.

*_1. Partnership PDF *_

We have a PDF of the partnership options (see attached). Ideally we can incorporate this on the partnership options page as a download (preferably after the page intro). It would be helpful for people to be able to download this PDF (and possibly email it), so that their ability to view the partnership options isn't limited to view on the web site.

**2. Individual Partners Page**

We need to create a page that is used to allow people to purchase Individual Sponsorships, and which also highlights our individual sponsorship options. You can see a good example on the BadCamp site here http://2012.badcamp.net/sponsors/individual and here: http://2013.badcamp.net/individual-sponsor . Collectively the individual sponsors were the largest overall sponsor of BADCamp, so there's clearly an lots of potential for this implementation to be of benefit to NYC Camp.

**3. Responsive Listing/Catalog of Options**

Currently the partnership options are displayed in a table which does not work well on mobile. Ideally we can make this more responsive. We can incorporate the stand alone pages to make for a more workable solution on mobile. Ultimately this page is really a product catalog and after the camp we should likely implement a more faceted search approach, and largely rely on the current table approach only for "Compare Partnership Options" functionality, similar to a typical e-commerce experience. As  such we shouldn't go to overboard with this right and try to keep the solution as simple/as quick as possible. For instance something like the https://drupal.org/project/responsive_tables module which hides columns other than price, program and table info is likely best, and then people click through to find out more.

**4. Stand Alone Pages for Partnership Options**

Ideally on the partnership options page, we can have each Partnership Option page link through to a stand alone page for individual Partnership Option. This means that we need to configure DS Layout for that page and theme it accordingly. It's important that this stand along page be responsive. For this is important for the Title/Tier Sponsorship Options, as it allows us to list out the benefits and and more details that is possible in the current table summary (e.g., including things like tweets schedule, press release schedule, photos from the prior year's partner partner pavilion, etc). This is also important for the Individual Sponsorship Options so that we can display more details of these options (eventually we can also ideally let individual sponsors choose amongst their swag -- e.g., T-Shirt vs Hat or Hoodie vs Bike Jersey) -- this would most likely be a custom line item (http://commerceguys.com/blog/introduction-custom-line-items-drupal-commerce). Overall however, the key benefit of the stand alone pages is making them responsive.

**5. Sponsorship Options Quick Preview**

We could potential use a Beauty Tips quick preview via hover or click to allow someone to get a quick snapshot of the extra details of the Partnership Option, without needing to actually visit the Individual Sponsorship options page. This would potentially allow us to reduce some of the clutter from the table summarizing the options.

*\* Shopping Cart **

Ideally we can implement a shopping cart block that appears as a static/expandable footer or header once a user has added something to their shopping cart. We should also likely consider some way of integrating this into the user block/menu that will eventually appear in the right hand column on the site (i.e., in addition to editing their profiles and a link to their schedule, a user has a View My Cart option).

Overall much of these are nice to have features for 2014, and things that we can work on more so for 2015. However, I wanted to post this summary in case anyone is keen on pushing the Commerce/Partnerships implementation forward. We can break these out into different issues on GitHub if thats preferred, but because this was all related I figured I'd group it together, and then we can divide and conquer with different issues if needed.

As were working through this we should also start to thing about how we want to implement partnerships year over year on the site (although let's not work on it now, just think about it), since that's going to be tricky to do.

There are a few more bit of functionality that we can add after this as a final step that will really help to make for a super commerce implementation and true showcase of how Drupal can deliver the 3Cs better than any other platform (Commerce + Community + Content) ... Drupal FTW!

**Comment by willykaram** [comment]
BTW, this is the partnership PDF that we want to include on the site: 

https://drive.google.com/file/d/0BxxoRYaHc6LJYS1HVERTTWR4Q1k/edit

**Comment by willykaram** [comment]
I updated the original list to also include a simple cart implementation -- sorry meant to include that in the original post.

**Comment by ForestMars** [comment]
On Mar 29, 2014, at 1:23 PM, Willy Karam notifications@github.com wrote:

> For instance something like the https://drupal.org/project/responsive_tables module

Speaking of, we should probably use https://drupal.org/project/views_responsive_grid for the session schedule. 

## 

"There are only two hard things in Computer Science: cache invalidation, naming things and off-by-one errors."
@forestmars
about.me/forestmars

**Comment by willykaram** [comment]
Good call. We should also use that on the partners page too (i.e., the wall
of logos).
